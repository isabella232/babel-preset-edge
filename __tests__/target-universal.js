import buildPreset from "../src"

import { check, fixtures, titles } from "./core"

describe("Target: Universal", () => {
  const options = buildPreset(null, {
    sourceMaps: false
  })

  fixtures.forEach((fileName, index) => {
    test(titles[index], () => check(fileName, options))
  })
})
