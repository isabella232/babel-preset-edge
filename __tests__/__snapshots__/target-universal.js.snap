// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Target: Universal comments 1`] = `
"import _camelCase from \\"lodash/camelCase\\";

function translate(value) {
  return _camelCase(value);
}"
`;

exports[`Target: Universal deadcode 1`] = `
"var STATIC = false;
before();
after();"
`;

exports[`Target: Universal emotion 1`] = `
"import { css as _css } from \\"emotion\\";

var className =
/*#__PURE__*/
_css({
  name: \\"1vdp66k-className\\",
  styles: \\"color:hotpink;label:className;\\"
});

function SomeComponent() {
  return React.createElement(\\"div\\", {
    className: className
  }, \\"Some hotpink text.\\");
}"
`;

exports[`Target: Universal es2015: classes 1`] = `
"import _inheritsLoose from \\"@babel/runtime/helpers/esm/inheritsLoose\\";
export var Component =
/*#__PURE__*/
function (_Other) {
  _inheritsLoose(Component, _Other);

  function Component() {
    var _this = _Other.call(this) || this;

    _this.flag = true;
    return _this;
  }

  var _proto = Component.prototype;

  _proto.main = function main() {
    alert(this.foo);
  };

  return Component;
}(Other);"
`;

exports[`Target: Universal es2015: for_off 1`] = `
"import \\"core-js/modules/es7.symbol.async-iterator\\";
import \\"core-js/modules/es6.symbol\\";
import \\"core-js/modules/web.dom.iterable\\";
import \\"core-js/modules/es6.array.is-array\\";

function test() {
  for (var _iterator = tasks, _isArray = Array.isArray(_iterator), _i = 0, _iterator = _isArray ? _iterator : _iterator[Symbol.iterator]();;) {
    var _ref;

    if (_isArray) {
      if (_i >= _iterator.length) break;
      _ref = _iterator[_i++];
    } else {
      _i = _iterator.next();
      if (_i.done) break;
      _ref = _i.value;
    }

    try {
      executeCommands();
    } catch (error) {
      console.error(\\"Error\\");
    }
  }
}"
`;

exports[`Target: Universal es2015: generator 1`] = `
"function* idMaker() {
  var index = 0;

  while (true) {
    yield index++;
  }
}"
`;

exports[`Target: Universal es2015: set 1`] = `
"import \\"core-js/modules/web.dom.iterable\\";
import \\"core-js/modules/es6.array.iterator\\";
import \\"core-js/modules/es6.string.iterator\\";
import \\"core-js/modules/es6.set\\";
new Set([1, 2, 3]);"
`;

exports[`Target: Universal es2016: exponentiation 1`] = `"var result = Math.pow(2, 3);"`;

exports[`Target: Universal esnext: async 1`] = `
"function wait() {
  return new Promise(function ($return, $error) {
    return Promise.resolve(loadData()).then(function () {
      try {
        return $return();
      } catch ($boundEx) {
        return $error($boundEx);
      }
    }, $error);
  });
}"
`;

exports[`Target: Universal esnext: async_loop 1`] = `
"import \\"core-js/modules/es7.symbol.async-iterator\\";
import \\"core-js/modules/es6.symbol\\";
import \\"core-js/modules/web.dom.iterable\\";
import \\"core-js/modules/es6.array.is-array\\";

function executeTasks(tasks) {
  return new Promise(function ($return, $error) {
    var _iterator, _isArray, _i, _ref;

    _iterator = tasks, _isArray = Array.isArray(_iterator), _i = 0, _iterator = _isArray ? _iterator : _iterator[Symbol.iterator]();
    var $Loop_2_trampoline;

    function $Loop_2() {
      if (_isArray) {
        if (_i >= _iterator.length) return [1];
        _ref = _iterator[_i++];
      } else {
        _i = _iterator.next();
        if (_i.done) return [1];
        _ref = _i.value;
      }

      var $Try_1_Post = function () {
        try {
          return $Loop_2;
        } catch ($boundEx) {
          return $error($boundEx);
        }
      },
          $Try_1_Catch = function () {
        try {
          console.error(\\"Error\\");
          return $Try_1_Post();
        } catch ($boundEx) {
          return $error($boundEx);
        }
      };

      try {
        return Promise.resolve(executeCommands()).then(function () {
          try {
            return $Try_1_Post();
          } catch ($boundEx) {
            return $Try_1_Catch($boundEx);
          }
        }, $Try_1_Catch);
      } catch (error) {
        $Try_1_Catch(error)
      }
    }

    return ($Loop_2_trampoline = function (q) {
      while (q) {
        if (q.then) return void q.then($Loop_2_trampoline, $error);

        try {
          if (q.pop) {
            if (q.length) return q.pop() ? $Loop_2_exit.call(this) : q;else q = $Loop_2;
          } else q = q.call(this);
        } catch (_exception) {
          return $error(_exception);
        }
      }
    }.bind(this))($Loop_2);

    function $Loop_2_exit() {
      return $return();
    }
  });
}"
`;

exports[`Target: Universal esnext: catch_bind 1`] = `
"try {
  throw 0;
} catch (_unused) {
  doSomethingWhichDoesntCareAboutTheValueThrown();
} finally {
  doSomeCleanup();
}"
`;

exports[`Target: Universal esnext: class properties 1`] = `
"export var Component = function Component() {
  this.onClick = function () {};
};"
`;

exports[`Target: Universal esnext: decorators 1`] = `
"var _class,
    MyClass = annotation(_class = function MyClass() {}) || _class;

function annotation(target) {
  target.annotated = true;
}"
`;

exports[`Target: Universal esnext: dynamic import 1`] = `
"import(
/*webpackChunkName:'lodash-wBYk8'*/
\\"lodash\\").then(function (lodash) {
  console.log(lodash.camelCase(\\"hello world\\"));
});"
`;

exports[`Target: Universal esnext: object spread 1`] = `
"import \\"core-js/modules/es6.object.assign\\";
var original = {
  bar: 10
},
    variant = Object.assign({
  foo: 1
}, original);"
`;

exports[`Target: Universal loadable 1`] = `
"import loadable from 'loadable-components';
export var Home = loadable({
  chunkName: function chunkName() {
    return \\"Home-JfPYC\\";
  },
  isReady: function isReady() {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },
  requireAsync: function requireAsync() {
    return import(
    /* webpackChunkName: \\"Home-JfPYC\\" */
    \\"./views/Home\\");
  },
  requireSync: function requireSync(props) {
    var id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },
  resolve: function resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/Home\\");
    }

    return require('path').resolve(__dirname, \\"./views/Home\\");
  }
});
export var About = loadable({
  chunkName: function chunkName() {
    return \\"About-8yDxR\\";
  },
  isReady: function isReady() {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },
  requireAsync: function requireAsync() {
    return import(
    /* webpackChunkName: \\"About-8yDxR\\" */
    \\"./views/About\\");
  },
  requireSync: function requireSync(props) {
    var id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },
  resolve: function resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/About\\");
    }

    return require('path').resolve(__dirname, \\"./views/About\\");
  }
});
export var Contact = loadable({
  chunkName: function chunkName() {
    return \\"Contact-AIf6m\\";
  },
  isReady: function isReady(props) {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },
  requireAsync: function requireAsync() {
    return import(
    /* webpackChunkName: \\"Contact-AIf6m\\" */
    \\"./views/Contact\\");
  },
  requireSync: function requireSync(props) {
    var id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },
  resolve: function resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/Contact\\");
    }

    return require('path').resolve(__dirname, \\"./views/Contact\\");
  }
});"
`;

exports[`Target: Universal loadable magic-comments 1`] = `
"import loadable from 'loadable-components';
export var Home = loadable({
  chunkName: function chunkName() {
    return \\"Home-JfPYC\\";
  },
  isReady: function isReady() {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },
  requireAsync: function requireAsync() {
    return import(
    /* webpackChunkName: \\"Home-JfPYC\\" */
    \\"./views/Home\\");
  },
  requireSync: function requireSync(props) {
    var id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },
  resolve: function resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/Home\\");
    }

    return require('path').resolve(__dirname, \\"./views/Home\\");
  }
});
export var About = loadable({
  chunkName: function chunkName() {
    return \\"About\\";
  },
  isReady: function isReady() {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },
  requireAsync: function requireAsync() {
    return import(
    /* webpackChunkName: \\"About\\" */
    \\"./views/About\\");
  },
  requireSync: function requireSync(props) {
    var id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },
  resolve: function resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/About\\");
    }

    return require('path').resolve(__dirname, \\"./views/About\\");
  }
});
export var Contact = loadable({
  chunkName: function chunkName() {
    return \\"Contact-AIf6m\\";
  },
  isReady: function isReady(props) {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },
  requireAsync: function requireAsync() {
    return import(
    /* webpackMode: \\"lazy\\", webpackFoo: \\"bar\\", webpackChunkName: \\"Contact-AIf6m\\" */
    \\"./views/Contact\\");
  },
  requireSync: function requireSync(props) {
    var id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },
  resolve: function resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/Contact\\");
    }

    return require('path').resolve(__dirname, \\"./views/Contact\\");
  }
});"
`;

exports[`Target: Universal loadable wildcard 1`] = `
"import loadable from 'loadable-components';
var page = \\"About\\",
    lang = \\"de_DE\\";
export var views = loadable({
  chunkName: function chunkName() {
    return \\"views-\\" + page;
  },
  isReady: function isReady() {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },
  requireAsync: function requireAsync() {
    return import(
    /* webpackChunkName: \\"views-[request]\\" */
    \\"./views/\\" + page);
  },
  requireSync: function requireSync(props) {
    var id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },
  resolve: function resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/\\" + page);
    }

    return require('path').resolve(__dirname, \\"./views/\\" + page);
  }
});
export var leanIntl = loadable({
  chunkName: function chunkName() {
    return \\"lean-intl-\\" + lang + \\"-data\\";
  },
  isReady: function isReady() {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },
  requireAsync: function requireAsync() {
    return import(
    /* webpackChunkName: \\"lean-intl-[request]-data\\" */
    \\"./lean-intl/\\" + lang + \\"/data\\");
  },
  requireSync: function requireSync(props) {
    var id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },
  resolve: function resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./lean-intl/\\" + lang + \\"/data\\");
    }

    return require('path').resolve(__dirname, \\"./lean-intl/\\" + lang + \\"/data\\");
  }
});
export var reactIntl = loadable({
  chunkName: function chunkName() {
    return \\"react-intl-\\" + lang + \\"-data\\";
  },
  isReady: function isReady(props) {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },
  requireAsync: function requireAsync() {
    return import(
    /* webpackChunkName: \\"react-intl-[request]-data\\" */
    \\"./react-intl/\\" + lang + \\"/data\\");
  },
  requireSync: function requireSync(props) {
    var id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },
  resolve: function resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./react-intl/\\" + lang + \\"/data\\");
    }

    return require('path').resolve(__dirname, \\"./react-intl/\\" + lang + \\"/data\\");
  }
});"
`;

exports[`Target: Universal lodash: async 1`] = `
"import _parallel from \\"async/parallel\\";

_parallel([]);"
`;

exports[`Target: Universal lodash: lodash 1`] = `
"import _camelCase from \\"lodash/camelCase\\";

_camelCase(\\"hello world\\");"
`;

exports[`Target: Universal lodash: recompose 1`] = `
"import _compose from \\"recompose/compose\\";

_compose([]);"
`;

exports[`Target: Universal macro: gql 1`] = `
"var query = {
  kind: \\"Document\\",
  definitions: [{
    kind: \\"OperationDefinition\\",
    operation: \\"query\\",
    name: {
      kind: \\"Name\\",
      value: \\"User\\"
    },
    variableDefinitions: [],
    directives: [],
    selectionSet: {
      kind: \\"SelectionSet\\",
      selections: [{
        kind: \\"Field\\",
        name: {
          kind: \\"Name\\",
          value: \\"user\\"
        },
        arguments: [{
          kind: \\"Argument\\",
          name: {
            kind: \\"Name\\",
            value: \\"id\\"
          },
          value: {
            kind: \\"IntValue\\",
            value: \\"5\\"
          }
        }],
        directives: [],
        selectionSet: {
          kind: \\"SelectionSet\\",
          selections: [{
            kind: \\"Field\\",
            name: {
              kind: \\"Name\\",
              value: \\"lastName\\"
            },
            arguments: [],
            directives: []
          }, {
            kind: \\"FragmentSpread\\",
            name: {
              kind: \\"Name\\",
              value: \\"UserEntry1\\"
            },
            directives: []
          }]
        }
      }]
    }
  }],
  loc: {
    start: 0,
    end: 79,
    source: {
      body: \\"\\\\n  query User {\\\\n    user(id: 5) {\\\\n      lastName\\\\n      ...UserEntry1\\\\n    }\\\\n  }\\\\n\\",
      name: \\"GraphQL request\\",
      locationOffset: {
        line: 1,
        column: 1
      }
    }
  }
};"
`;

exports[`Target: Universal macro: preval 1`] = `"var one = 1;"`;

exports[`Target: Universal promise 1`] = `
"import \\"core-js/modules/es6.promise\\";
export default function () {
  return new Promise(function (resolve) {
    return resolve(42);
  });
}"
`;

exports[`Target: Universal react: jsx 1`] = `
"function render() {
  return React.createElement(\\"h1\\", null, \\"Hello\\");
}"
`;

exports[`Target: Universal react: jsx spread 1`] = `
"import \\"core-js/modules/es6.object.assign\\";

function render(props) {
  return React.createElement(\\"h1\\", Object.assign({
    className: \\"large\\"
  }, props), \\"Hello\\");
}"
`;

exports[`Target: Universal react: proptypes 1`] = `
"import _inheritsLoose from \\"@babel/runtime/helpers/esm/inheritsLoose\\";
import React from \\"react\\";
import PropTypes from \\"prop-types\\";

var MyComponent =
/*#__PURE__*/
function (_React$Component) {
  _inheritsLoose(MyComponent, _React$Component);

  function MyComponent() {
    return _React$Component.apply(this, arguments) || this;
  }

  var _proto = MyComponent.prototype;

  _proto.render = function render() {
    return React.createElement(\\"div\\", null, this.props.userName);
  };

  return MyComponent;
}(React.Component);

MyComponent.propTypes = {
  userName: PropTypes.string
};"
`;

exports[`Target: Universal require context 1`] = `
"if (process.env.NODE_ENV === \\"test\\") {
  require('babel-plugin-require-context-hook/register')();
}

var loader = require.context(__dirname, false, /\\\\.js/);"
`;
